autompc/costs/cost.py

```diff
@@ -186,6 +186,12 @@ def goal(self) -> Optional[np.ndarray]:
         return np.copy(self.properties['goal'])
     
     @goal.setter
+    def goal(self, goal):
```

Comment:
To use @goal.setter property decorator , the function name must be the same as the attribute's name. To be consistent with the coding style of other files, I kept the original Java-style setter function in lines 195-197